docs/DafnyRef/UserGuide.md

```diff
@@ -166,9 +167,27 @@ A few options are not part of a command. In these cases any single-hyphen spelli
 
 The `dafny resolve` command checks the command-line and then parses and typechecks the given files and any included files.
 
-Note that a complete program must be presented to the dafny tool, either by listing all files on the command-line, or by using `include` directives,
-or by some combination. An incomplete program will likely result in errors because of unresolved names. A program need not include all implementations
-of all functions in order to verify parts of it, but will need all implementations in order to compile a working executable.
+The set of files considered by `dafny` are those listed on the command-line,
+including those named in a `--library` option, and all files that are
+named, recursively, in `include` directives in files in the set being considered by the tool.
+
+The set of files presented to an invocation of the `dafny` tool must 
+contain all the declarations needed to resolve all names and types, 
+else name or type resolution errors will be emitted.
```

Comment:
I think "or else" or "otherwise" is more idiomatic.

Reply:
I'll change it along with future other edits.